scripts/zones/Arrapago_Reef/npcs/qm1.lua

```diff
@@ -3,17 +3,17 @@
 --  NPC: ??? (Spawn Lil'Apkallu(ZNM T1))
 -- !pos 488 -1 166 54
 -----------------------------------
-local ID = require("scripts/zones/Arrapago_Reef/IDs");
+local ID = require("scripts/zones/Arrapago_Reef/IDs")
+require("scripts/globals/npc_util")
+-----------------------------------
 
 function onTrade(player,npc,trade)
-    if (trade:hasItemQty(2601,1) and trade:getItemCount() == 1) then -- Trade Greenling
-        if (not GetMobByID(ID.mob.LIL_APKALLU):isSpawned()) then
-            player:tradeComplete();
-            SpawnMob(ID.mob.LIL_APKALLU):updateClaim(player);
-        end
+            if npcUtil.tradeHas(trade, 2601) and npcUtil.popFromQM(player, npc, ID.mob.LIL_APKALLU) then
```

Comment:
![Image of TabSoda](http://www.drinkwhat.com/wp-content/uploads/2011/01/Coca-Cola-Tab-Diet-Soda-Drink.jpg)